lib/protoflow-blocks/src/blocks/sys/write_socket.rs

```diff
@@ -0,0 +1,185 @@
+extern crate std;
+use super::SysBlocks;
+use crate::prelude::{vec, String};
+use crate::{IoBlocks, StdioConfig, StdioError, StdioSystem, System};
+use core::str::FromStr;
+use protoflow_core::{
+    Block, BlockError, BlockResult, BlockRuntime, InputPort, Message, SystemBuilding,
+};
+use protoflow_derive::Block;
+use serde::{Deserialize, Serialize};
+use simple_mermaid::mermaid;
+use std::{
+    format,
+    net::TcpStream,
+    sync::{Arc, Mutex, PoisonError},
+};
+use tracing::error;
+/// A block that writes a proto object to a TCP port.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/sys/write_socket.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/sys/write_socket.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     // TODO
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute WriteSocket host="127.0.0.1" port="7077" buffer_size="1024"
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct WriteSocket<T: Message> {
```

Comment:
There is no need for this block to be doing encoding as well. It really should just write bytes to the socket, being on par and compatible with `WriteStdout`, `WriteStderr`, `WriteFile`, and so on.

Reply:
done